### PR TITLE
texstudio: 4.5.1 -> 4.5.2

### PR DESCRIPTION
**Pull request by MrTarantoga**
###### Description of changes

Version update and change build system, because qmake is not the current build system. They switched to cmake.
See more in [changelog](https://texstudio-org.github.io/CHANGELOG.html).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/texstudio/default.nix [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-    hash = "sha256-QBPWelCqh8Ggp0IjH37QGYGu5Ya2fpsiSEkKh3Ee7PM=";
+    hash = "sha256-L+q4+k4XWywnxIvhfXPCBeCqnGj4E9BhAmgcAZhm7S8=";
   };
 
-  nativeBuildInputs = [ qmake wrapQtAppsHook pkg-config ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook pkg-config ];
   buildInputs = [ qtbase qtscript qtsvg poppler zlib ];
 
-  qmakeFlags = [ "NO_APPDATA=True" ];
+  cmakeFlags = [ "NO_APPDATA=True" ];
```
Comment (wegank) [inline pkgs/applications/editors/texstudio/default.nix:18]: `NO_APPDATA=True` does not seem to be a valid flag. Is the line still necessary?

Comment (MrTarantoga) [reply pkgs/applications/editors/texstudio/default.nix:18]: Your right. I'll remove it.

Comment (MrTarantoga) [reply pkgs/applications/editors/texstudio/default.nix]: See https://github.com/NixOS/nixpkgs/pull/230531/commits/5eb977e81830665c0a83d22ce5e2b3e8893882d4

Comment (wegank) [reply pkgs/applications/editors/texstudio/default.nix]: Please squash the commits, thanks!

Comment (MrTarantoga) [reply pkgs/applications/editors/texstudio/default.nix]: Done

**Review by wegank — APPROVED** [review]

**Pull request merged**